src/main/scala/com/fulcrumgenomics/testing/VariantContextSetBuilder.scala

```diff
@@ -0,0 +1,163 @@
+    this._header.addMetaDataLine(headerLine)
+    this
+  }
+
+  /** Add a variant. */
+  def addVariant(refIdx: Int = 0,
+                 start: Long,
+                 variantAlleles: List[String],
+                 genotypeAlleles: List[String] = List.empty,
+                 sampleName: Option[String] = None,
+                 phased: Boolean = false): this.type = {
+    val contig          = this.dict.getSequence(refIdx).getSequenceName
+    val alleles         = toAlleles(variantAlleles)
+    val stop            = VariantContextUtils.computeEndFromAlleles(alleles, start.toInt, -1)
+    val (ctxBuilder, prevGenotypes) = this.variants.find { ctx =>
+      ctx.getContig == contig &&
+        ctx.getStart == start &&
+        ctx.getEnd == stop &&
+        ctx.getAlleles.map(_.getBaseString).sorted.mkString("") == variantAlleles.sorted.mkString("") // only an approximation
+    } match {
+      case Some(ctx) =>
+        this.variants = this.variants.filter(c => c != ctx)
+        (new VariantContextBuilder(ctx), ctx.getGenotypesOrderedByName.toList)
+      case None      => (new VariantContextBuilder("source", contig, start, stop, alleles), List.empty[Genotype])
+    }
+    val referenceAllele = ctxBuilder.getAlleles.find(_.isReference)
+    val genotypeBuilder = sampleName match {
+      case Some(sn) => new GenotypeBuilder(sn, if (genotypeAlleles.isEmpty) ctxBuilder.getAlleles else toAlleles(genotypeAlleles, referenceAllele=referenceAllele))
+      case None     => new GenotypeBuilder(this._header.getSampleNamesInOrder.head, ctxBuilder.getAlleles)
+    }
+    genotypeBuilder.phased(phased)
+    val genotypes = genotypeBuilder.make() +: prevGenotypes
+    ctxBuilder.genotypes(genotypes:_*)
+    this.variants.append(ctxBuilder.make())
+    this
+  }
+
+  /** Gets an iterator over the variants. */
+  override def iterator = this.variants.iterator
+
+  /** Returns the number of variants added to this builder. */
+  override def size: Int = this.variants.size
+
+  /** Writes the contents of the record set to the provided file path. */
+  def write(path: PathToVcf) : Unit = {
+    val writer = new VariantContextWriterBuilder()
+      .setReferenceDictionary(this.dict)
+      .setOutputFile(path.toFile)
+      .setOption(Options.INDEX_ON_THE_FLY)
+      .setOption(Options.ALLOW_MISSING_FIELDS_IN_HEADER)
+      .build()
+    writer.writeHeader(_header)
+    this.iterator.foreach(writer.add)
+    writer.close()
+  }
+
+  /** Writes the contents to a temporary file that will be deleted when the JVM exits. */
+  def toTempFile(deleteOnExit: Boolean = true): PathToVcf = {
+    val path = Files.createTempFile("VariantContextSet.", ".vcf.gz")
+    if (deleteOnExit) path.toFile.deleteOnExit()
+    write(path)
+    path
+  }
+
+  /** Writes the contents to a temporary file that will be deleted when the JVM exits and
+    * provides a [VCFFileReader]. */
+  def toVCFFileReader(deleteOnExit: Boolean = true): VCFFileReader = {
```

Comment:
I know the type is this way, but can we have the method be `toVcfFileReader` please?